rust/datafusion/src/datasource/memory.rs

```diff
@@ -365,4 +366,59 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_merged_schema() -> Result<()> {
+        let mut metadata = HashMap::new();
+        metadata.insert("foo".to_string(), "bar".to_string());
+
+        let schema1 = Schema::new_with_metadata(
+            vec![
+                Field::new("a", DataType::Int32, false),
```

Comment:
I think this test should also include different numbers of fields in the two record batches (e.g. remove "c" from the first batch and remove "b" from the second)